api/buckets/client/options.go

```diff
@@ -5,6 +5,36 @@ import (
 	"github.com/ipfs/interface-go-ipfs-core/path"
 )
 
+type initOptions struct {
+	name         string
+	private      bool
+	bootstrapCid cid.Cid
+}
+
+type InitOption func(*initOptions)
+
+// WithName sets a name for the bucket.
+func WithName(name string) InitOption {
+	return func(args *initOptions) {
+		args.name = name
+	}
+}
+
+// WithPrivate sets an encryption password for the bucket.
```

Comment:
```suggestion
// WithPrivate specifies that an encryption password will be used for the bucket.
```

Reply:
Not sure about that, but the language here suggests WithPrivate is _setting_ the password, as opposed to setting a flag.

Reply:
Nice, I like it.